drkafka/src/main/java/com/pinterest/doctorkafka/replicastats/ReplicaStatsManager.java

```diff
@@ -90,29 +94,18 @@ private static long getLastReplicaReassignmentTimestamp(String brokerZkUrl,
   public static void update(BrokerStats brokerStats) {
     String brokerZkUrl = brokerStats.getZkUrl();
     // ignore the brokerstats from clusters that are not enabled operation automation.
-    if (!config.getClusterZkUrls().contains(brokerZkUrl)) {
+    if (clusterZkUrls == null) {
+      clusterZkUrls = config.getClusterZkUrls();
+    }
+    if (!clusterZkUrls.contains(brokerZkUrl)) {
       return;
     }
 
-    KafkaCluster cluster;
-    synchronized (clusters) {
-      if (brokerZkUrl != null && !clusters.containsKey(brokerZkUrl)) {
-        DoctorKafkaClusterConfig clusterConfig = config.getClusterConfigByZkUrl(brokerZkUrl);
-        assert clusterConfig != null;
-
-        cluster = new KafkaCluster(brokerZkUrl, clusterConfig);
-        clusters.put(brokerZkUrl, cluster);
-      }
-      cluster = clusters.get(brokerZkUrl);
-    }
+    KafkaCluster cluster = clusters.computeIfAbsent(brokerZkUrl, url -> new KafkaCluster(url, config.getClusterConfigByZkUrl(url)));
```

Comment:
if a broker is down, kafkastats agent can still send  brokerstat  back.  do we need to check `brokerZkUrl == null`  in that case?

Reply:
should be captured by L100, will comment this issue there